src/requirements/Farcaster/hooks/useFarcasterCast.ts

```diff
@@ -0,0 +1,37 @@
+import useDebouncedState from "hooks/useDebouncedState"
+import useSWRImmutable from "swr/immutable"
+
+const BASE_URL =
+  "https://api.neynar.com/v2/farcaster/cast?api_key=NEYNAR_API_DOCS&identifier="
+
+const useFarcasterCast = (hash?: string, url?: string) => {
+  const debouncedHash = useDebouncedState(hash)
+  const debouncedUrl = useDebouncedState(url)
+
+  const fetchUrl = debouncedHash
```

Comment:
Consider validating the hash & url here, becuase I managed to run into this issue, hehe:
<img width="395" alt="image" src="https://github.com/guildxyz/guild.xyz/assets/92519134/d85c0705-5c39-453b-855a-d6deaccaefb5">
(we shouldn't fetch anything if we pass invalid data)